program/rust/src/tests/pyth_simulator.rs

```diff
@@ -139,7 +102,19 @@ struct ProductMetadata {
 impl PythSimulator {
     /// Deploys the oracle program as upgradable
     pub async fn new() -> PythSimulator {
-        let mut bpf_data = read_file(&*ORACLE_PROGRAM_BINARY_PATH);
+        let mut cmd = std::process::Command::new("cargo");
```

Comment:
I think the right way of solving this is replacing `cargo test` by `cargo-test-bpf` in `build-bpf.sh`.
That will internally call `cargo-build-bpf` before starting the tests.